source/Octopus.Tentacle/Commands/MigratePreInstalledKubernetesDeploymentTargetCommand.cs

```diff
@@ -0,0 +1,119 @@
+            Options.Add("namespace=", "The namespace to use for the migration", v => @namespace = v);
+        }
+        
+        // This command is only used as a way to programatically copy the config map and secret from the pre-installation hook to the new agent
+        // It does not access tentacle configuration so that it doesn't 
+        protected override void Start()
+        {
+            if (!PlatformDetection.Kubernetes.IsRunningAsKubernetesAgent)
+            {
+                throw new ControlledFailureException("This command can only be run from within a Kubernetes agent.");
+            }
+
+            // Check that the sources and destinations are different
+            if (sourceSecretName == destinationSecretName || sourceConfigMapName == destinationConfigMapName)
+            {
+                log.Error("Source and destination names must be different.");
+                return;
+            }
+            
+            var migrationNamespace = @namespace ?? KubernetesConfig.Namespace;
+            
+            var config = configProvider.Value.Get();
+            var client = new k8s.Kubernetes(config);
+            var sourceConfigMap = TryGetCoreV1Object(() => client.CoreV1.ReadNamespacedConfigMap(sourceConfigMapName, migrationNamespace));
+            var sourceSecret = TryGetCoreV1Object(() => client.CoreV1.ReadNamespacedSecret(sourceSecretName, migrationNamespace));
+            var destinationConfigMap = TryGetCoreV1Object(() => client.CoreV1.ReadNamespacedConfigMap(destinationConfigMapName, migrationNamespace));
+            var destinationSecret = TryGetCoreV1Object(() => client.CoreV1.ReadNamespacedSecret(destinationSecretName, migrationNamespace));
+
+            var (migrateData, reason) = ShouldMigrateData(sourceConfigMap, sourceSecret, destinationConfigMap, destinationSecret);
+            if (!migrateData)
+            {
+                log.Info(reason);
+                return;
+            }
+
+            // Copy the data from the source to the destination
+            destinationConfigMap!.Data = sourceConfigMap!.Data;
+            destinationSecret!.Data = sourceSecret!.Data;
+            client.CoreV1.ReplaceNamespacedConfigMap(destinationConfigMap, destinationConfigMapName, migrationNamespace);
+            client.CoreV1.ReplaceNamespacedSecret(destinationSecret, destinationSecretName, migrationNamespace);
+            
+            // Delete the sources (they are no longer needed)
+            client.CoreV1.DeleteNamespacedConfigMap(sourceConfigMapName, migrationNamespace);
+            client.CoreV1.DeleteNamespacedSecret(sourceSecretName, migrationNamespace);
+            
+            log.Info("Migration complete.");
+        }
+        
+        public static (bool, string) ShouldMigrateData(V1ConfigMap? sourceConfigMap, V1Secret? sourceSecret, V1ConfigMap? destinationConfigMap, V1Secret? destinationSecret)
```

Comment:
Not sure what version of c# we're on here but, if possible, this will make the tests easier to read.

```suggestion
        public static (bool ShouldMigrate, string Reason) ShouldMigrateData(V1ConfigMap? sourceConfigMap, V1Secret? sourceSecret, V1ConfigMap? destinationConfigMap, V1Secret? destinationSecret)
```